test/common/config/grpc_subscription_test_harness.h

```diff
@@ -34,21 +35,30 @@ namespace Config {
 
 class GrpcSubscriptionTestHarness : public SubscriptionTestHarness {
 public:
-  GrpcSubscriptionTestHarness() : GrpcSubscriptionTestHarness(std::chrono::milliseconds(0)) {}
+  GrpcSubscriptionTestHarness(Envoy::Config::LegacyOrUnified legacy_or_unified)
+      : GrpcSubscriptionTestHarness(legacy_or_unified, std::chrono::milliseconds(0)) {}
 
-  GrpcSubscriptionTestHarness(std::chrono::milliseconds init_fetch_timeout)
+  GrpcSubscriptionTestHarness(Envoy::Config::LegacyOrUnified legacy_or_unified,
+                              std::chrono::milliseconds init_fetch_timeout)
       : method_descriptor_(Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
             "envoy.service.endpoint.v3.EndpointDiscoveryService.StreamEndpoints")),
-        async_client_(new NiceMock<Grpc::MockAsyncClient>()) {
+        async_client_(new NiceMock<Grpc::MockAsyncClient>()),
+        should_use_unified_(legacy_or_unified == Envoy::Config::LegacyOrUnified::Unified) {
     node_.set_id("fo0");
     EXPECT_CALL(local_info_, node()).WillRepeatedly(testing::ReturnRef(node_));
     ttl_timer_ = new NiceMock<Event::MockTimer>(&dispatcher_);
 
     timer_ = new Event::MockTimer(&dispatcher_);
 
-    mux_ = std::make_shared<Config::GrpcMuxImpl>(
-        local_info_, std::unique_ptr<Grpc::MockAsyncClient>(async_client_), dispatcher_,
-        *method_descriptor_, random_, stats_store_, rate_limit_settings_, true);
+    if (legacy_or_unified == Envoy::Config::LegacyOrUnified::Unified) {
```

Comment:
nit:
```suggestion
    if (should_use_unified_) {
```
might be more readable. Leaving it up to you